index.rst

```diff
@@ -1,61 +1,338 @@
 
-   Feel free to delete this instructional comment.
+IDM-0003
+    Prospective users may authenticate through another federated identity provider.
+    In this case, their pending account request will be held for approval by an existing user with authority to approve new accounts.
+    This flow will be used to handle collaborators who would not otherwise have the necessary affiliation to create an account.
 
-:tocdepth: 1
+IDM-0004
+    Once a user has created an account, they may attach additional federated identities to the same account.
+    This should require authentication via one of their current identity providers and then via the new identity provider.
+    Once this has been done, both identities will map to the same account in the Science Platform.
+    This allows users with multiple institutional affiliations to use whichever authentication provider is most convenient for them.
+    [DMS-LSP-REQ-0024]
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+IDM-0005
+    Authenticated users must be able to view all federated identities attached to their account and revoke any of them.
 
-.. sectnum::
+IDM-0006
+    GitHub and Google must be supported as additional identity providers.
+    They may be added as additional identities to an existing account via IDM-0004, or be used by collaborators as described in IDM-0003.
+
+IDM-0007
+    Users with administrative access to the Science Platform (for example, the ability to impersonate other users) must authenticate with two authentication factors.
+    These will generally a password or some equivalent, and some additional security token bound to a sandboxed device (not a general-purpose laptop), such as an OTP authenticator on a mobile phone.
+    Since all authentication is via federated identity (IDM-0001), this must be implemented by requiring authentication from an external provider that, in turn, requires two-factor authentication.
+    GitHub or Google (IDM-0005), with appropriate configuration, would be suitable.
+
+IDM-0008
+    The authentication session generated by a web authentication must follow reasonable practices for secure cookie storage.
+    This means, at least, setting ``Secure`` and ``HttpOnly`` and using either opaque session tokens or cryptographically-signed data.
+
+IDM-0009
+    If a user primarily uses non-affiliation-providing identity providers such as Google and GitHub, there must be a mechanism to force the user to reauthenticate using an affiliation-providing identity provider to reconfirm their affiliation.
+    This shouldn't be done routinely, but may be necessary periodically to confirm that a user still has an appropriate affiliation.
+    One possible mechanism to implement this may be to add an expiration date for the user that is automatically extended when the user logs on with an appropriate identity provider.
+
+IDM-0010
+    Users may select their own account name.
```

Comment:
Is this something you see happening during the sign up flow?  I'm not even sure if we have a signup flow now, since I don't think I remember going through it.  Being able to pick an account name seems reasonable, although since this gets into paths and could percolate into services (especially if they are able to change their name) we should probably be pretty clear about that and what might need to happen if a username changes (or conflicts with another one, since it's federated, we could have name collisions)?

Reply:
Yup, this would happen during the sign-up flow. The username is unrelated to their federated identity. I'll make that clear.